src/lib.rs

```diff
@@ -254,3 +265,334 @@ impl<'a, NP: ?Sized + NixPath>  NixPath for Option<&'a NP> {
         }
     }
 }
+
+/*
+ *
+ * ===== Null terminated slices for exec =====
+ *
+ */
+
+use std::ops::{Deref, DerefMut};
+use std::mem::transmute;
+use std::{iter, io};
+
+/// An error returned from the [`TerminatedSlice::from_slice`] family of
+/// functions when the provided data is not terminated by `None`.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct NotTerminatedError {
```

Comment:
That seems like unnecessary bloat to `NixError` for an error case specific to a single function call that no one will ever actually use or call in 99% of scenarios.

Also chances are if someone does use it, they'll already have created the data to be terminated and actually want the `unsafe` variant anyway (or if you're avoiding unsafe then failure of the call will be treated as an internal failure that's just `unwrap()`'d or `unreachable!()`). I can't really imagine a scenario where this error would ever be propagated up a method chain or actually used with `try!()`.

Reply:
I'd make the argument that it's less bloat than all the code implementing a new error type, but I understand your point.

As to why this should be an `Error` type is because the semantics behind that better match what's happening rather than `Option`. If there is a failure, it should be an error. `Option`s aren't used to denote failure. There is a grey area where it can go either way, but I think this is a clear case for `Error.`

So let's go ahead and leave this error as is. A goal of mine is to unify the error handling within `nix` to use the new `failure` crate, so this will likely get cleaned up along with that change.